### DISCUSSION TITLE
How to customize footnote component in `@astro/mdx`?

### DISCUSSION THREAD

**Discussion by ngdangtu-vn** [remarkjs/remark#1366] [Q&A]
According to [the document](https://mdxjs.com/docs/using-mdx/#components), I can do something like this:

```jsx
<Example
    components={{
      // Map `h1` (`# heading`) to use `h2`s.
      h1: 'h2',
      // Rewrite `em`s (`*like so*`) to `i` with a goldenrod foreground color.
      em(props) {
        return <i style={{color: 'goldenrod'}} {...props} />
      },
      // Pass a layout (using the special `'wrapper'` key).
      wrapper({components, ...rest}) {
        return <main {...rest} />
      },
      // Pass a component.
      Planet() {
        return 'Neptune'
      },
      // This nested component can be used as `<theme.text>hi</theme.text>`
      theme: {
        text(props) {
          return <span style={{color: 'grey'}} {...props} />
        }
      }
    }}
  />
```

It's great I'm looking for a way to customize the footnote components like about, I try with a few names and none of them work. Does anyone know how to overwrite the footnote components? I need to rewrite a whole html structure of these components (assume it has 2)

btw, it is not ideal to overwrite the html tag such as `sup` or `a` as it can be used for other purposes (I even want to replace `section` tag as well)

**Comment by wooorm** [comment]
I recommend not doing what you want to do.

Turning footnotes from markdown into HTML is done by https://github.com/remarkjs/remark-rehype. See its docs.

**Reply by ngdangtu-vn** [reply]
so do we have option to disable footnote only?

**Reply by wooorm** [reply]
Please use more words.
I think no.
But it is very hard to understand you.
